sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamSuite.scala

```diff
@@ -1079,6 +1079,42 @@ class StreamSuite extends StreamTest {
       assert(query.exception.isEmpty)
     }
   }
+
+  Seq(true, false).foreach { useV2Sink =>
+    import org.apache.spark.sql.functions._
+
+    val newTestName = "SPARK-26379 Structured Streaming - Exception on adding column to Dataset" +
+      s" - use v2 sink - $useV2Sink"
+
+    test(newTestName) {
+      val input = MemoryStream[Int]
+      val df = input.toDS().withColumn("cur_timestamp", lit(current_timestamp()))
```

Comment:
As I see this covers `CurrentTimestamp`, maybe worth to add test for `CurrentDate` as well.